src/chainstate/stacks/boot/costs-2.clar

```diff
@@ -1 +1,567 @@
+    {
+        runtime: (linear n u81 u1303),
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: (linear n u1 u1)
+    })
+
+
+(define-read-only (cost_analysis_fetch_contract_entry (n uint))
+    {
+        runtime: (linear n u1000 u1000),
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: (linear n u1 u1)
+    })
+
+(define-read-only (cost_lookup_variable_depth (n uint))
+    (runtime (linear n u2 u14)))
+
+(define-read-only (cost_lookup_variable_size (n uint))
+    (runtime (linear n u2 u1)))
+
+(define-read-only (cost_lookup_function (n uint))
+    (runtime u16))
+
+(define-read-only (cost_bind_name (n uint))
+    (runtime u256))
+
+(define-read-only (cost_inner_type_check_cost (n uint))
+    (runtime (linear n u2 u9)))
+
+(define-read-only (cost_user_function_application (n uint))
+    (runtime (linear n u26 u140)))
+
+(define-read-only (cost_let (n uint))
+    (runtime (linear n u146 u862)))
+
+(define-read-only (cost_if (n uint))
+    (runtime u200))
+
+(define-read-only (cost_asserts (n uint))
+    (runtime u158))
+
+(define-read-only (cost_map (n uint))
+    (runtime (linear n u1210 u3314)))
+
+(define-read-only (cost_filter (n uint))
+    (runtime u460))
+
+(define-read-only (cost_len (n uint))
+    (runtime u486))
+
+(define-read-only (cost_element_at (n uint))
+    (runtime u619))
+
+(define-read-only (cost_index_of (n uint))
+    (runtime (linear n u1 u243)))
+
+(define-read-only (cost_fold (n uint))
+    (runtime u483))
+
+(define-read-only (cost_list_cons (n uint))
+    (runtime (linear n u14 u198)))
+
+(define-read-only (cost_type_parse_step (n uint))
+    (runtime u5))
+
+(define-read-only (cost_tuple_get (n uint))
+    (runtime (nlogn n u4 u1780)))
+
+(define-read-only (cost_tuple_merge (n uint))
+    (runtime (linear n u4 u646)))
+
+(define-read-only (cost_tuple_cons (n uint))
+    (runtime (nlogn n u11 u1101)))
+
+(define-read-only (cost_add (n uint))
+    (runtime (linear n u12 u156)))
+
+(define-read-only (cost_sub (n uint))
+    (runtime (linear n u12 u156)))
+
+(define-read-only (cost_mul (n uint))
+    (runtime (linear n u14 u157)))
+
+(define-read-only (cost_div (n uint))
+    (runtime (linear n u14 u157)))
+
+(define-read-only (cost_geq (n uint))
+    (runtime u166))
+
+(define-read-only (cost_leq (n uint))
+    (runtime u166))
+
+(define-read-only (cost_le (n uint))
+    (runtime u166))
+
+(define-read-only (cost_ge (n uint))
+    (runtime u166))
+
+(define-read-only (cost_int_cast (n uint))
+    (runtime u164))
+
+(define-read-only (cost_mod (n uint))
+    (runtime u168))
+
+(define-read-only (cost_pow (n uint))
+    (runtime u170))
+
+(define-read-only (cost_sqrti (n uint))
+    (runtime u167))
+
+(define-read-only (cost_log2 (n uint))
+    (runtime u161))
+
+(define-read-only (cost_xor (n uint))
+    (runtime u167))
+
+(define-read-only (cost_not (n uint))
+    (runtime u162))
```

Comment:
Shouldn't this one be cheaper than `log2` ?

Reply:
I'm not sure -- `log2` is a pretty fast operation on integers: just a bitshift, `not` is fast too, but I'm not sure its easy to say which should be faster.

Reply:
`not` inverses boolean value so it changes `true` to `false` or the other way around. It operates on 1bit.
It should be the cheapest or one of the cheapest functions that we have in Clarity.